src/core/collection/Collection.tsx

```diff
@@ -294,10 +294,24 @@ export function EventSelectionCollection({
   });
 
   // Reduce past events that are no longer available and therefore do not need to be displayed
-  const eventsList = data?.eventsByIds.data.filter(
+  const eventsListFiltered = data?.eventsByIds.data.filter(
     (event) => !isEventClosed(event),
   );
 
+  const eventsListSorted = [];
+
+  // sorting events in the same order it was defined in cms
+  if (eventsListFiltered?.length > 0) {
+    collection.events.forEach((eventId) => {
+      const event = eventsListFiltered.find(
+        (eventData) => eventData.id === eventId,
+      );
+      if (event) {
+        eventsListSorted.push(event);
+      }
+    });
+  }
+
```

Comment:
I would use this syntax instead `itemsArray.sort((a, b) => sortingArr.indexOf(a) - sortingArr.indexOf(b));`.

So, I think this would become something like this 

```
  const eventsList = data?.eventsByIds.data
    .filter((event) => !isEventClosed(event))
    .sort((a, b) => collection.events.indexOf(a.id) - collection.events.indexOf(b.id));
```